platform/mellanox/mlnx-platform-api/sonic_platform/fan.py

```diff
@@ -208,32 +262,70 @@ def set_status_led(self, color):
             return False
         status = False
         try:
-            if color == 'green':
+            if color == self.STATUS_LED_COLOR_GREEN:
                 with open(os.path.join(LED_PATH, self.fan_green_led_path), 'w') as fan_led:
                     fan_led.write(str(LED_ON))
-            elif color == 'red':
+                    status = True
+            elif color == self.STATUS_LED_COLOR_RED:
                 # Some fan don't support red led but support orange led, in this case we set led to orange
-                if 'red' in led_cap_list:
+                if self.STATUS_LED_COLOR_RED in led_cap_list:
                     led_path = os.path.join(LED_PATH, self.fan_red_led_path)
-                elif 'orange' in led_cap_list:
+                elif self.STATUS_LED_COLOR_ORANGE in led_cap_list:
                     led_path = os.path.join(LED_PATH, self.fan_orange_led_path)
                 else:
                     return False
                 with open(led_path, 'w') as fan_led:
                     fan_led.write(str(LED_ON))
-
-            elif color == 'off':
-                with open(os.path.join(LED_PATH, self.fan_green_led_path), 'w') as fan_led:
-                    fan_led.write(str(LED_OFF))
-
-                with open(os.path.join(LED_PATH, self.fan_red_led_path), 'w') as fan_led:
-                    fan_led.write(str(LED_OFF))
+                    status = True
+            elif color == self.STATUS_LED_COLOR_OFF:
+                if self.STATUS_LED_COLOR_GREEN in led_cap_list:
+                    with open(os.path.join(LED_PATH, self.fan_green_led_path), 'w') as fan_led:
+                        fan_led.write(str(LED_OFF))
+                if self.STATUS_LED_COLOR_RED in led_cap_list:
+                    with open(os.path.join(LED_PATH, self.fan_red_led_path), 'w') as fan_led:
+                        fan_led.write(str(LED_OFF))
+                if self.STATUS_LED_COLOR_ORANGE in led_cap_list:
+                    with open(os.path.join(LED_PATH, self.fan_orange_led_path), 'w') as fan_led:
+                        fan_led.write(str(LED_OFF))
+
+                status = True
             else:
                 status = False
         except (ValueError, IOError):
-                    status = False
+            status = False
+
         return status
 
+
+    def get_status_led(self):
+        """
+        Gets the state of the fan status LED
+
+        Returns:
+            A string, one of the predefined STATUS_LED_COLOR_* strings above
+        """
+        led_cap_list = self._get_led_capability()
+        if led_cap_list is None:
+            return self.STATUS_LED_COLOR_OFF
+
+        try:
+            with open(os.path.join(LED_PATH, self.fan_green_led_path), 'r') as fan_led:
+                if '0' != fan_led.read().rstrip('\n'):
```

Comment:
shall we use '1' == ?  and mabe can also define LED_ON and LED_OFF here as psu led

Reply:
don't use '1'. '0' stands for off and other value stands for 'on'
will use LED_ON and LED_OFF instand of magic numbers

Reply:
will use LED_ON and LED_OFF instand of magic numbers
but won't use '1' since '0' stands for off and other value stands for 'on'